dashboard/app/entities.go

```diff
@@ -514,3 +517,93 @@ func timeDate(t time.Time) int {
 	year, month, day := t.Date()
 	return year*10000 + int(month)*100 + day
 }
+
+type RecipientType int
+
+const (
+	To RecipientType = iota
+	Cc
+)
+
+// t.String() must not be changed as the data is stored in this format.
+func (t RecipientType) String() string {
+	return [...]string{"To", "Cc"}[t]
+}
+
+type RecipientInfo struct {
+	Address mail.Address
+	Type    RecipientType
+}
+
+type Recipients []RecipientInfo
+
+func (r Recipients) Len() int           { return len(r) }
+func (r Recipients) Less(i, j int) bool { return r[i].Address.Address < r[j].Address.Address }
+func (r Recipients) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+
+type RecipientString string
+
+func (r Recipients) String() RecipientString {
+	recStrSlice := RecipientStrSlice(r.StringSlice())
+	return RecipientString(strings.Join(recStrSlice, "|"))
+}
+
+func RecipientStrSlice(emails []RecipientString) []string {
```

Comment:
Is there any point in having the `RecipientString` type at all when it results in having to do these extra conversions to/from raw strings?

Why not just convert directly between the string in the database and `Recipients`?